File: Java/Classes/Offers.java
public class Offers extends RequestDonationList {
    private final Organization organization;

    public Offers(Organization organization) {
        super();
        this.organization = organization;
    }

    @Override
    public void add(RequestDonation rd, Organization organization) throws Exceptions.EntityNotFoundException {
        super.add(rd, organization);
    }

    // In Offers.java
    public void commit() throws Exceptions.EntityNotFoundException {
        System.out.println("Starting commit process..."); // Add debug line
        for (RequestDonation offer : getRdEntities()) {
            RequestDonation currentDonation = organization.getCurrentDonation(offer.getEntity());
            System.out.println("Processing offer: " + offer.getEntity().getName() +
                    " - Quantity: " + offer.getQuantity()); // Add debug line

            if (currentDonation != null) {
                double newQuantity = currentDonation.getQuantity() + offer.getQuantity();
                currentDonation.setQuantity(newQuantity);
                System.out.println("Updated quantity: " + newQuantity); // Add debug line
            } else {
                organization.addCurrentDonation(new RequestDonation(offer.getEntity(), offer.getQuantity()));
                System.out.println("Added new donation"); // Add debug line
            }
        }
        reset();
        System.out.println("Commit completed"); // Add debug line
    }
}

File: C++/cpp_files/Offers.cpp
// Offers.cpp
#include "Offers.h"
#include "exceptions.h"
#include "Organization.h"
#include <iostream>

Offers::Offers() : donator(nullptr) {
}

Offers::~Offers() = default;

void Offers::setDonator(Donator* don) {
    donator = don;
}

void Offers::commit() {
    if (rdEntities.empty()) {
        return;  // Nothing to commit
    }

    try {
        // Update or add each offer to currentDonations
        for (const auto& rd : rdEntities) {
            auto entityId = rd->getEntity()->getId();
            auto quantity = rd->getQuantity();

            try {
                // Try to get current quantity and update it
                RequestDonation* currentRd = organization->getRequestDonation(entityId);
                if (currentRd) {
                    double newQuantity = currentRd->getQuantity() + quantity;
                    organization->updateCurrentDonation(entityId, newQuantity);
                } else {
                    organization->addCurrentDonation(entityId, quantity);
                }
            } catch (const EntityNotFoundException&) {
                // If entity not found, add as new donation
                organization->addCurrentDonation(entityId, quantity);
            }
        }

        // Clear the list after successful commit
        reset();
    } catch (const std::exception& e) {
        throw CommitFailedException("Failed to commit offers: " + std::string(e.what()));
    }
}